README.md

```diff
@@ -73,19 +88,39 @@ for its content.
 ***NUR does not check repository for malicious content on a regular base and it is
 recommend to check expression before installing them.***
 
+### Using modules overlays or library functions on NixOS
+
+If you intend to use modules, overlays or library functions in your NixOS configuration.nix, you need to take care to not introduce infinite recursion. Specifically, you need to import NUR like this in the modules:
+
+```nix
+{ pkgs, config, lib, ... }:
+let
+  nur = import (builtins.fetchTarball "https://github.com/nix-community/NUR/archive/master.tar.gz") {};
```

Comment:
It is easy to shadow nur by using `environment.systemPackages = [ nur.repos.mic92.inxi ];` instead of `environment.systemPackages = [ pkgs.nur.repos.mic92.inxi ];`. 
In this case `pkgs` in NUR will be undefined and the error message you added is thrown.
Maybe the example should not called it `nur` to avoid this namespace collision?
The alternative could be also to make `nur` a nixos module (also it is probably not possible to dynamically inject imports and overlays?) - maybe a function that returns a nixos module could work.

Reply:
> The alternative could be also to make nur a nixos module

Yeah I thought of that! Would actually solve this problem rather elegantly: Users could add something like `imports = [ (import <nur/modules> {}) ]` (replace nur with the fetch command) exactly once to their nixos config.

To make this work, NUR should wrap every repositories modules under an enable option. So e.g. when I now want to use your foo module (which you declared in `modules.foo`), I'd do `nur.mic92.foo.enable = true`. This would work like this, and I might implement this in a future PR. For now this more hacky solution should be fine, especially since it can be forwards compatible.

Reply:
Oh and regarding the `nur` attribute name, I think changing it to `nur_for_modules` would work well.

Reply:
True, this can be also implemented in future. I also intentionally put repos as sub namespace so we can put more into the top-level hierarchy.

Reply:
Tracked here: https://github.com/nix-community/NUR/issues/43

Reply:
`s/nur_for_modules/nur-modules/g`

Reply:
Maybe `nur-no-pkgs`? Because the example there also uses overlays with it.

Reply:
Agreed.